pkg/collector/corechecks/oracle-dbm/custom_queries.go

```diff
@@ -0,0 +1,153 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build oracle
+
+package oracle
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	godror "github.com/godror/godror"
+	"reflect"
+	"strconv"
+)
+
+type Method func(string, float64, string, []string)
+
+type metricRow struct {
+	name   string
+	value  float64
+	method Method
+	tags   []string
+}
+
+func (c *Check) CustomQueries() error {
+	if c.dbCustomQueries == nil {
+		db, err := c.Connect()
+		if err != nil {
+			if errClosing := CloseDatabaseConnection(db); err != nil {
+				log.Errorf("Error closing connection %s", errClosing)
+			}
+			return err
+		}
+		if db == nil {
+			return fmt.Errorf("empty connection")
+		}
+		c.dbCustomQueries = db
+		c.dbCustomQueries.SetMaxOpenConns(1)
+	}
+
+	var metricRows []metricRow
+	sender, err := c.GetSender()
+	if err != nil {
+		return fmt.Errorf("failed to get sender for custom queries %w", err)
+	}
+	methods := map[string]Method{"gauge": sender.Gauge, "count": sender.Count, "rate": sender.Rate}
+	for _, q := range c.config.CustomQueries {
+		var errInQuery bool
+		metricPrefix := q.MetricPrefix
+
+		if metricPrefix == "" {
+			log.Error("Undefined metric_prefix for a custom query")
+			continue
+		}
+		log.Tracef("custom query configuration %v", q)
+		var pdb string
+		if !c.connectedToPdb {
+			pdb = q.Pdb
+			if pdb == "" {
+				pdb = "cdb$root"
+			}
+			_, err := c.dbCustomQueries.Exec(fmt.Sprintf("alter session set container = %s", pdb))
+			if err != nil {
+				log.Errorf("Can't set container to %s", pdb)
+				continue
+			}
+		}
+		rows, err := c.dbCustomQueries.Queryx(q.Query)
+		if rows != nil {
+			defer rows.Close()
+		}
+		if err != nil {
+			log.Errorf("failed to fetch rows for the custom query %s %s", metricPrefix, err)
+			continue
+		}
+		var metricsFromSingleRow []metricRow
+		for rows.Next() {
+			tags := []string{fmt.Sprintf("pdb:%s", pdb)}
+			cols, err := rows.SliceScan()
+			if err != nil {
+				log.Errorf("failed to get values for the custom query %s %s", metricPrefix, err)
+				errInQuery = true
+				break
+			}
+			if len(cols) > len(q.Columns) {
+				log.Errorf("Not enough column mappings for the custom query %s %s", metricPrefix, err)
+				errInQuery = true
+				break
+			}
+			var metricRow metricRow
+			for i, v := range cols {
+				if q.Columns[i].Type == "tag" {
+					if v != nil {
+						tags = append(tags, fmt.Sprintf("%s:%s", q.Columns[i].Name, v))
+					}
+				} else if methodFunc, ok := methods[q.Columns[i].Type]; ok {
+					metricRow.name = fmt.Sprintf("%s.%s", metricPrefix, q.Columns[i].Name)
+					if v_str, ok := v.(string); ok {
+						metricRow.value, err = strconv.ParseFloat(v_str, 64)
+						if err != nil {
+							log.Errorf("Metric value %v for metricRow.name %s is not a number", v, metricRow.name)
```

Comment:
These are user-facing logs, right? If I was a user and saw this line in the logs, I probably would not know how to respond. This log with no context doesn't tell me much:

`2023-07-05 15:06:53 UTC | CORE | INFO | (pkg/collector/oracle-dbm/custom_queries.go:46 in RunAll) | Metric value octo for metricRow.name cat is not a number`

I would prefer to see additional information and context like this: 

`2023-07-05 15:06:53 UTC | CORE | INFO | (pkg/collector/oracle-dbm/custom_queries.go:46 in RunAll) | Custom query #1 encountered a type error during execution. A number was expected for the column "name" but the query results returned the value "octo" (query was: "SELECT user1, user2, ...")`

Reply:
I improved the error messages.